`hint_normal_roughness_texture` sampler2D causes shader compiler error in GLES3/Compatibility mode

### Godot version

4.0.beta16

### System information

Windows 10 Pro x64, GLES3, RTX 3080 Ti (528.02)

### Issue description

In Compatibility/GLES3 back-end, shaders which use a uniform to get the normal roughness buffer cannot compile. For example, the shader:
```glsl
shader_type spatial;

uniform sampler2D normal_roughness_texture : hint_normal_roughness_texture;

void fragment() {
    vec4 sample = texture(normal_roughness_texture, UV);
}
```
results in an error in the output:
```
  drivers/gles3/shader_gles3.cpp:242 - SceneShaderGLES3: Fragment shader compilation failed:
  0(890) : error C1503: undefined variable "normal_roughness_buffer"
  
  drivers/gles3/shader_gles3.cpp:342 - Method/function failed.
```

If I switch over to the Mobile or Forward+ backends, there is no shader compilation error.

### Steps to reproduce

Open the attached reproduction in the 4.0.beta16 editor.

When the scene is opened, the error is produced automatically:
1. Open `scene.tscn`
2. Open Output to review error

When the scene is saved with any changes made to the shader, the same error is also produced:
1. Open `scene.tscn`
2. Open `example.gdshader` in the shader editor.
3. Make a compile-able change to the shader.
4. Save the scene
5. Open Output to review the error

### Minimal reproduction project

[NormalRoughnessTexture_GLES3_Error.zip](https://github.com/godotengine/godot/files/10534253/NormalRoughnessTexture_GLES3_Error.zip)

--- a/servers/rendering/shader_language.cpp
+++ b/servers/rendering/shader_language.cpp
@@ -8759,6 +8759,10 @@ Error ShaderLanguage::_parse_shader(const HashMap<StringName, FunctionInfo> &p_f
 									new_hint = ShaderNode::Uniform::HINT_NORMAL_ROUGHNESS_TEXTURE;
 									--texture_uniforms;
 									--texture_binding;
+									if (OS::get_singleton()->get_current_rendering_method() == "gl_compatibility") {
+										_set_error(RTR("'hint_normal_roughness_texture is not supported in gl_compatibility shaders."));
+										return ERR_PARSE_ERROR;
+									}
 								} break;
 								case TK_HINT_DEPTH_TEXTURE: {
 									new_hint = ShaderNode::Uniform::HINT_DEPTH_TEXTURE;

